src/main/java/biz/paluch/spinach/output/JobOutput.java

```diff
@@ -35,13 +39,24 @@ public void set(ByteBuffer bytes) {
             return;
         }
 
-        body = codec.decodeValue(bytes);
+        if (body == null) {
+            counters = new HashMap<String, Long>();
+            body = codec.decodeValue(bytes);
+            return;
+        }
+
+        lastKey = decodeAscii(bytes);
+    }
+
+    @Override
+    public void set(long integer) {
+        counters.put(lastKey, integer);
```

Comment:
Hrm - true, though the current protocol shouldn't do that. Perhaps something like `if (lastKey != null) put and reset lastKey`?